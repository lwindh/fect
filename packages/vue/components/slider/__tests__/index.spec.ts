import { Slider } from '..'
import { mount } from '@vue/test-utils'
import { trigger } from '../../../tests'

describe('Slider', () => {
  it('should render as normal', () => {
    const wrapper = mount(Slider, {
      props: {
        modelValue: 0
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('props should work correctly', () => {
    const wrapper = mount(Slider, {
      props: {
        hideValue: true,
        showMarkers: true,
        step: 5,
        disabled: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should trigger event when click', async () => {
    const wrapper = mount(Slider, {
      props: { modelValue: 0 }
    })
    const el = wrapper.find('.fect-slider')

    await el.trigger('click')
    await wrapper.setProps({ disabled: true })
    await el.trigger('click')
    expect(wrapper.html()).toMatchSnapshot()
  })

  // it('should trigger events when drag', async () => {
  //   const wrapper = mount(Slider, {
  //     props: { modelValue: 0 },
  //   })
  //   const el = wrapper.find('.fect-slider__dot')
  //   await trigger('touchstart', el, 0, 0)
  //   await trigger('touchmove', el, 50, 0)
  //   await trigger('touchend', el, 50, 0)
  //   await expect(wrapper.html()).toMatchSnapshot()
  // })
})
