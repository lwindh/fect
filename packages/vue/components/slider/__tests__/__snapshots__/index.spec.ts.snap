// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider props should work correctly 1`] = `
<div class="fect-slider disabled" role="slider" title="0%">
  <div class="fect-slider__dot " style="left: 0%;">
    <!---->
  </div><span class="fect-slider__mark" style="left: 5%;"></span><span class="fect-slider__mark" style="left: 10%;"></span><span class="fect-slider__mark" style="left: 15%;"></span><span class="fect-slider__mark" style="left: 20%;"></span><span class="fect-slider__mark" style="left: 25%;"></span><span class="fect-slider__mark" style="left: 30%;"></span><span class="fect-slider__mark" style="left: 35%;"></span><span class="fect-slider__mark" style="left: 40%;"></span><span class="fect-slider__mark" style="left: 45%;"></span><span class="fect-slider__mark" style="left: 50%;"></span><span class="fect-slider__mark" style="left: 55%;"></span><span class="fect-slider__mark" style="left: 60%;"></span><span class="fect-slider__mark" style="left: 65%;"></span><span class="fect-slider__mark" style="left: 70%;"></span><span class="fect-slider__mark" style="left: 75%;"></span><span class="fect-slider__mark" style="left: 80%;"></span><span class="fect-slider__mark" style="left: 85%;"></span><span class="fect-slider__mark" style="left: 90%;"></span><span class="fect-slider__mark" style="left: 95%;"></span>
</div>
`;

exports[`Slider should render as normal 1`] = `
<div class="fect-slider " role="slider" title="0%">
  <div class="fect-slider__dot " style="left: 0%;">0</div>
  <!---->
</div>
`;

exports[`Slider should trigger event when click 1`] = `
<div class="fect-slider disabled" role="slider" title="0%">
  <div class="fect-slider__dot click" style="left: 0%;">0</div>
  <!---->
</div>
`;
